feat: add QA testcase vagrant box definitions, scripts and examples

tcooper opened the pull request:
This PR adds capability to conduct QA Testcases in semi-automated or manual fashion.

The following testcases are covered:

- [QA:Testcase Mediacheck](https://wiki.rockylinux.org/team/testing/qa_test_cases/)
- [QA:Testcase Debranding Analysis](https://wiki.rockylinux.org/team/testing/qa_test_cases/)

The style of this submission serves as an example layout for other possible QA:Testcase scenarios but is not the only style that may be submitted and/or accepted.

---
## Author checklist (to be completed by original Author)

- [x] I have signed the Rocky Open Source Contributor Agreement (ROSCA).
- [x] I certify this contribution complies with all conditions of the
  [Testing repository contributions guide](./CONTRIBUTING.md)
- [x] If applicable, steps and instructions have been tested to work on a real
  system.
- [x] I have installed pre-commit and all commits in this PR pass.
- [x] If applicable, I have GPG signed all commits and unploaded my private key
  to Github.

---
## Rocky Testing checklist  (to be completed by Rocky team)

- [x] 1st Pass (Check that contribution is good fit for project and the Testing
  repository is the appropriate location)
- [x] 2nd Pass (Confirm all pre-commit tests pass)
- [ ] 3nd Pass (Technical Review - check for technical correctness)

tcooper committed:
update .gitignore for qa-testcase-boxes

tcooper committed:
basic layout for qa-testcase-boxes

tcooper committed:
add testcase-mediacheck

tcooper committed:
add testcase-mediacheck sample output

tcooper committed:
add testcase-mediacheck examples

tcooper committed:
add testcase-debranding-validation

tcooper committed:
add testcase-debranding-validation sample output

tcooper committed:
add testcase-debranding-validation examples

akatch commented on qa-testcase-boxes/testcase-debranding-validation/peridot_yumrepofs.yml:
```diff
@@ -0,0 +1,7 @@
+---
```
How come the keys in this yaml file are quoted?

tcooper replied:
Has to do with trying to get `yq` to read all the keys, mostly the `dotnet3.0` one which I ended up transliterating to `dotnet3_0` in the code anyway. The rest is history (ie. copy/paste). It's valid yaml and passes yamllint even if not commonly used.

akatch replied:
Got it, thanks for the explanation!

tcooper committed:
rename script and provide basic documentation

tcooper committed:
update script and documentation

tcooper committed:
update debranding configuration for r9

tcooper committed:
document process for updating debranding configuration

tcooper committed:
update debranding configuration for r9

tcooper committed:
abrt is removed in r9. move from all section to r8 section

tcooper committed:
ensure repository metadata is current

tcooper committed:
update debranding configuration for r9

tcooper commented:
No longer a draft. All packages on the debrand list for r9 pass debranding from Peridot yumrepofs local cache.

tcooper commented:
Rebased to `development` branch to use as example with new branch protections. A single review should unblock this PR from merge.

StackKorora commented:
There are a bunch of output files. Are those necessary to be in the git repo or can that folder be excluded?

mkahric committed:
Add Rocky Linux 9 RC1 test reports (#77)

* Add Rocky Linux 9 RC1 test reports

VirtualBox 6.1 (Windows 7 64-bit), UEFI without Secure Boot;

* Add Rocky Linux 9 RC1 test reports

Workstation 15.5 Player (Windows 7 64-bit), UEFI with Secure Boot

* Add Rocky Linux 9 RC1 test reports

Hyper-V 10 (Windows 10 64-bit), UEFI with Secure Boot

StackKorora committed:
Adding a few dev Intel systems to the Rocky 9 xsos reports (#78)

* Adding a few dev Intel systems.
* Fixing my goof; Needs to be in the rl9 subdir.

tcooper commented:
Output files are there as examples and are intentionally included but could be removed.

dithotxgh committed:
xsos for RL9 install Dell Precision_Workstation_T3500 (#79)

Co-authored-by: Doug Maxey <pops@maxeygroup.tech>

tcooper committed:
Merge remote-tracking branch 'upstream/main' into qa-testcase-boxes

tcooper committed:
update r8.yaml for 8.7

tcooper committed:
update r9.yaml for 9.0

tcooper committed:
updates for 8.7 (tested with 9.0)

tcooper committed:
package version updates for 9.1

tcooper committed:
file target and/or content changes for 9.1

tcooper committed:
remove open-vm-tools for r9

tcooper committed:
remove sos from r8 and r9

tcooper committed:
add lorax for r9

tcooper committed:
add osbuild for r9

tcooper committed:
add gnome-session for r9

tcooper committed:
remove chrony for r8 and r9

tcooper committed:
update peridot projectid for r9.1

tcooper committed:
add support for r9 DVD media

tcooper committed:
powertools in r8, crb in r9

tcooper committed:
ensure fresh repository metadata

tcooper commented:
Went to create new PR for the recent testing and found this PR was still unmerged. Question from @StackKorora about output files which were provided as examples. Response was they could be removed and no more discussion. Will bring up in next meeting...

tcooper committed:
don't add/show some files in git status

tcooper committed:
update debranding yaml for Rocky Linux 9.2

tcooper committed:
update peridot yumrepofs for Rocky Linux 9.2

tcooper committed:
pre-9.5 sync

tcooper committed:
pre-9.5 fixes

tcooper committed:
pre-9.6 update

tcooper committed:
more pre-9.6 updates

tcooper committed:
various ISO download and mediacheck updates

tcooper committed:
remove circular ref

tcooper committed:
change default box for testcase-mediacheck to r9

tcooper committed:
change default box for testcase-mediacheck to rockylinux/9

tcooper committed:
change default box for testcase-debrand-validation to rockylinux/9

tcooper committed:
Merge branch 'qa-testcase-boxes' of github.com:tcooper/rocky-linux-testing into qa-testcase-boxes

tcooper committed:
deprecate support for r8 and add support for r10 in testcase-mediacheck

tcooper committed:
r9 updates for testcase-debrand-validation

tcooper committed:
r9.6 updates for testcase-debrand-validation

tcooper committed:
r10 additions for testcase-debrand-validation

tcooper committed:
r10.1 updates for testcase-debrand-validation

tcooper committed:
canonicalize common_opts.sh for testcase-mediacheck

tcooper commented:
@StackKorora updates to this branch for newer releases since your comment that I said we would discuss but never did. I think it's fine to remove the bulk of the output files but also believe it's useful to keep a small set as examples documentation purposes. I have the changes from 9.7 to add before getting ready for the next releases. If you could add some specificity related to your previous comment and confirm / deny that you agree with my thoughts on examples I can get this repo into a state where review will be easier. Also, I'm going to remove Lukas and Al as named reviewers since they haven't been active doing reviews for Testing team for some time.

tcooper committed:
CHECKSUM signature files return .asc file extention

tcooper committed:
update patch.yml parsing

tcooper committed:
update Vagrantfile

tcooper committed:
update configuration for r9.8

tcooper committed:
update config for r10.2

tcooper committed:
update .gitignore
